### ISSUE TITLE
[ANDROID] Feature with empty string id causing NumberFormatException

### ISSUE THREAD

**Issue by sddamico** [mapbox/mapbox-gl-native#9594]
**Platform:** Android
**Mapbox SDK version:** 5.1.1

### Steps to trigger behavior

 1. Add shape features with an empty string id to map (not sure if possible with runtime styling, but certainly possible with vector tiles)
 2. Tap on shape...

```
07-24 14:49:03.443 15223-15223/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                   Process: com.seatgeek.android, PID: 15223
                                                   java.lang.NumberFormatException: For input string: ""
                                                       at java.lang.Long.parseLong(Long.java:455)
                                                       at java.lang.Long.valueOf(Long.java:542)
                                                       at com.mapbox.mapboxsdk.maps.AnnotationManager$ShapeAnnotationHitResolver.execute(AnnotationManager.java:729)
                                                       at com.mapbox.mapboxsdk.maps.AnnotationManager.onTap(AnnotationManager.java:651)
                                                       at com.mapbox.mapboxsdk.maps.MapGestureDetector$GestureListener.onSingleTapConfirmed(MapGestureDetector.java:318)
                                                       at android.view.GestureDetector$GestureHandler.handleMessage(GestureDetector.java:300)
                                                       at android.os.Handler.dispatchMessage(Handler.java:102)
                                                       at android.os.Looper.loop(Looper.java:154)
                                                       at android.app.ActivityThread.main(ActivityThread.java:6121)
                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
```

**Comment by jfirebaugh** [comment]
Duplicate of #9584.

**Comment by tobrun** [comment]
@sddamico, 
The issue is fixed with https://github.com/mapbox/mapbox-gl-native/pull/9585 but I'm still trying to understand the root cause. ShapeAnnotationHitResolver should only query layers with id:

`com.mapbox.annotations.shape.i`

Where `i` equals the id of shape annotation. Could you validate if your vector tile has such layers?

**Comment by tobrun** [comment]
This issue was figured out, when you don't have any ShapeAnnotations, it will query all the layers.

**Comment by sddamico** [comment]
@tobrun awesome, glad you all were able to find the root cause there.

Any chance we might see an accelerated release to get this out? 5.1.1 is a no-go for us with this issue...

**Comment by tobrun** [comment]
We are aiming at doing 2 weekly patch releases, at the end of this week fixes will be cherry-picked and snaphot will be generated (this should unblock you for development). A final 5.1.2 should happen somewhere next week.

**Comment by tobrun** [comment]
@sddamico 5.1.2-SNAPSHOT is available for testing
